cpu/esp_common/syscalls.c

```diff
@@ -142,13 +156,17 @@ void IRAM_ATTR _lock_acquire(_lock_t *lock)
 
     assert(lock != NULL);
 
+    uint32_t state = irq_disable();
```

Comment:
This should not be needed, the `mutex_lock()` below should disable IRQs at critical sections already

Reply:
This is what I thought too. But without disabling interrupts, `tests/malloc_thread_safe` crashes on ESP32.